src/test/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5ClientTest.java

```diff
@@ -379,10 +386,35 @@ void GIVEN_client_WHEN_update_subscriptions_with_null_message_handler_THEN_throw
     }
 
     @Test
-    void GIVEN_client_with_subscription_request_WHEN_retryable_reason_code_received_THEN_subscription_will_retry
+    void GIVEN_client_WHEN_unsubscribe_from_topic_with_retryable_reason_code_THEN_retry_unsubscribe
             (ExtensionContext context) throws RetryableMqttOperationException {
         ignoreExceptionOfType(context, RetryableMqttOperationException.class);
         String topic = "iotcore/topic";
+        String topic2 = "iotcore/topic2";
+        Set<String> topics = new HashSet<>();
+        topics.add(topic);
+        LocalMqtt5Client clientSpy = spy(client);
+
+        // subscribe to topics
+        clientSpy.updateSubscriptions(topics, message -> {});
+
+        topics.remove(topic);
+        topics.add(topic2);
+        mockMqtt5Client.nextUnsubAckReasonCode.add(topic, UnsubAckPacket.UnsubAckReasonCode.UNSPECIFIED_ERROR);
+
+        // subscribe to new topics, this will unsubscribe from the old topics
+        clientSpy.updateSubscriptions(topics, message -> {});
+        assertThat("subscribed topics local client", clientSpy::getSubscribedLocalMqttTopics,
+                eventuallyEval(is(topics)));
```

Comment:
why eventually? why won't this be true immediately?

Reply:
It is true immediately, the rest of the tests use eventually so I followed that, but I can update those as well

Reply:
Just spoke with Joe about this, we're using eventually to future proof the tests in case we decide to make the clients fully async later on